src/main/java/wolox/training/services/OpenLibrary.java

```diff
@@ -24,55 +29,94 @@
     @Autowired
     private Environment env;
 
+    @Autowired
+    private BookRepository bookRepository;
+
     public OpenLibrary(RestTemplateBuilder restTemplateBuilder) {
         restTemplate = restTemplateBuilder.build();
     }
 
-    private JsonElement doIsbnRequest(String isbn) throws Exception {
+    private String doIsbnRequest(String isbn) throws IOException {
         String baseUrl = env.getProperty("services.openLibraryBooksBaseUrl");
         String url = baseUrl + "/books?bibkeys=ISBN:" + isbn + "&format=json&jscmd=data";
         URL urlPath = new URL(url);
         HttpURLConnection con =(HttpURLConnection) urlPath.openConnection();
         con.setRequestMethod("GET");
-        if (con.getResponseCode() != HttpURLConnection.HTTP_OK)
-            return null;
+        int code = con.getResponseCode();
+        if (code == HttpURLConnection.HTTP_NOT_FOUND) {
+            throw new EntityNotFoundException(Book.class);
+        }
+        if (code != HttpURLConnection.HTTP_OK) {
+            throw new IOException();
```

Comment:
The approach is great! The only thing that I would change is the exception, IOException is used for input/output. Maybe you can have your own exception. Something like `ExternalServiceException`, which could receive the name of the failing external service

Reply:
:D